### PR TITLE
Enable Mastodon Apps: Implement the account endpoint for an external account

### PR DESCRIPTION
**Pull request by akirk**
This moves functionality from the Enable Mastodon Apps into ActivityPub. It allows retrieving the account of someone via ActivityPub with an API endpoint like `/api/v1/accounts/alex@kirk.at/`

The unit test currently needs network access, we should add the ability to mock the response.

### CODE REVIEW

File: integration/class-enable-mastodon-apps.php [hunk 1]
```diff
@@ -71,4 +83,54 @@ function ( $item ) {
...
+		}
+
+		$acct = Webfinger_Util::uri_to_acct( $uri );
+		$data = get_remote_metadata_by_actor( $uri );
+
+		if ( ! $data ) {
+			return $user_data;
+		}
+
+		$account = new Account();
```
Comment (pfefferle) [inline integration/class-enable-mastodon-apps.php:113]: To not always replace the complete we should re-use `$user_date` if from type Account. What do you think?

```suggestion
        if ( $user_data instanceOf 'Account' ) {
         		$account = $user_data;
        } else {
        		$account = new Account();
        }
```

**Review by pfefferle — DISMISSED** [review]
thanks @akirk

File: integration/class-enable-mastodon-apps.php [hunk 2]
```diff
@@ -71,4 +83,58 @@ function ( $item ) {
...
+
+	/**
+	 * Resolve external accounts for Mastodon API
+	 *
+	 * @param Enable_Mastodon_Apps\Entity\Account $user_data The user data
+	 * @param string                              $user_id   The user id
+	 *
+	 * @return Enable_Mastodon_Apps\Entity\Account The filtered Account
+	 */
+	public static function api_account_external( $user_data, $user_id ) {
```
Comment (pfefferle) [inline integration/class-enable-mastodon-apps.php:95]: what about renaming it to `api_account` and to extend also the internal accounts with `acct` and follower counts? Or should we do that in a different hook?

**Review by pfefferle — APPROVED** [review]
👍

File: integration/class-enable-mastodon-apps.php (hunk 2 above)
Comment (akirk) [reply integration/class-enable-mastodon-apps.php:95]: Added that second hook in https://github.com/Automattic/wordpress-activitypub/pull/676

**Pull request merged**